api/scripts/mypy_check.py

```diff
@@ -0,0 +1,74 @@
+import re
+import subprocess
+import sys
+from pathlib import Path
+
+
+def main() -> None:
+    baseline_path = Path("scripts/mypy_baseline.txt")
+    if not baseline_path.exists():
+        print("Baseline file not found. Run mypy and create mypy_baseline.txt first.")
+        sys.exit(1)
+
+    # Run mypy and capture output
+    files_to_check = [f"../{filename}" for filename in sys.argv[1:]]
+    command = [
+        "poetry",
+        "run",
+        "mypy",
+        "--config-file",
+        "pyproject.toml",
+    ] + files_to_check
+
+    result = subprocess.run(
+        command,
+        capture_output=True,
+        text=True,
+    )
+
+    # Mypy found no issues for the targeted file
+    if result.returncode == 0:
+        print(result.stdout.strip())
+        sys.exit(0)
+
+    # Mypy failed in some other way other than listing failing file lines
+    if result.returncode != 1:
+        print(command)
+        print(
+            f"Error running mypy with return code {result.returncode} and error:",
+            result.stderr,
+        )
+        print(result.stdout.strip())
```

Comment:
These 3 print statements seem overkill? Particularly the first one that prints out the command - I suspect that is left from the debugging that you did around the pre-commit files being passed in? 👀

Reply:
Yeah I was thinking that it could be useful for someone to debug the mypy error without running pre-commit but in retrospect that probably isn't necessary. I've removed the `print(command)` line. The other two should still live there though.